### PR TITLE
NUTS paper benchmarks

### PR DESCRIPTION
**Pull request by trappmartin**
This adds the old NUTS benchmark test we had in [link](https://github.com/TuringLang/TuringExamples/tree/old-models/old-models) back to the benchmark suit.

This depends on #826 for the Stochastic volatility model to work efficiently.

### CODE REVIEW

File: benchmarks/nuts/hlr.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Sampling parameter settings
+n_samples = 10_000
+n_adapts = 1_000
+
+# Sampling
+bench_res = @tbenchmark_expr("NUTS(Leapfrog(...))",
+                             sample(hlr_nuts(x, y, 1/0.01),
+                             NUTS(n_samples, n_adapts, 0.65)));
+
+LOG_DATA = build_log_data("[NUTS] HierarchicalLogisticRegression-Benchmark", bench_res...)
```
Comment (KDr2) [inline benchmarks/nuts/hlr.jl]: We have no `build_log_data` now, we could directly assign the return value of `@tbenchmark_expr` or `@tbenchmark` to `LOG_DATA`:

```julia
LOG_DATA = @tbenchmark_expr("NUTS(Leapfrog(...))",
                             sample(hlr_nuts(x, y, 1/0.01),
                             NUTS(n_samples, n_adapts, 0.65)));
```

File: benchmarks/nuts/lr.jl [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        y[n] ~ BinomialLogit(1, dot(x[n,:], β) + α)
+    end
+end
+
+# Sampling parameter settings
+n_samples = 10_000
+n_adapts = 1_000
+
+# Sampling
+bench_res = @tbenchmark_expr("NUTS(Leapfrog(...))",
```
Comment (KDr2) [inline benchmarks/nuts/lr.jl]: ditto: `LOG_DATA = @tbench....`

File: benchmarks/nuts/sv_nuts.jl [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    end
+end
+
+
+# Sampling parameter settings
+n_samples = 10_000
+n_adapts = 1_000
+
+# Sampling
+bench_res = @tbenchmark_expr("NUTS(Leapfrog(...))",
```
Comment (KDr2) [inline benchmarks/nuts/sv_nuts.jl]: ditto.

**Comment by trappmartin** [comment]
I removed the respective `build_log_data` calls and adjusted the codes.

**Comment by trappmartin** [comment]
By merging with the master, I must have somehow added commits in the docs that I did not intend to add. I'll check why that happened and reverse this before merging.

**Comment by yebai** [comment]
@trappmartin perhaps rebase master branch into this PR? It seems this PR contains a lot of code not from benchmarking purposes.

**Comment by trappmartin** [comment]
Yes, I think I accidentally merged some version of the master into this PR. I'll try to rebase asap.

**Comment by trappmartin** [comment]
WTF

**Pull request closed without merging**